### PR TITLE
[BUG] RandomResizedCrop throws runtime error when called with p=0

### PR DESCRIPTION
**Pull request by FGeri**
### Description
This PR fixes the issue that RandomResizedCrop throws runtime error when called with p=0

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/augmentation/random_generator/random_generator.py [hunk 1]
```diff
@@ -354,6 +354,9 @@ def random_crop_size_generator(
     _joint_range_check(scale, "scale")
     _joint_range_check(ratio, "ratio")
 
+    if batch_size == 0:
+        return dict(size=torch.zeros([0, 2]))
```
Comment (shijianjian) [inline kornia/augmentation/random_generator/random_generator.py:358]: Why returning [0, 2]? Maybe returning an empty tensor?

It should not be running according to:

https://github.com/kornia/kornia/blob/master/kornia/augmentation/base.py#L161-L163

Comment (FGeri) [reply kornia/augmentation/random_generator/random_generator.py:358]: It's gonna be an empty tensor.
Right, but the parameter generation will earlier and https://github.com/kornia/kornia/blob/fc4b5a452110478e67de3f0a257e18ed036e18e7/kornia/augmentation/random_generator/random_generator.py#L370-L371 will crash.

Comment (shijianjian) [reply kornia/augmentation/random_generator/random_generator.py:358]: I see. I do not think returning random stuff is appropriate here.

How about something like:
```python
if batch_size != 0:
   w_out = w[torch.arange(0, batch_size), torch.argmax(cond, dim=1)] 
   h_out = h[torch.arange(0, batch_size), torch.argmax(cond, dim=1)] 
else:
   w_out = w
   h_out = hd
```

The idea is we should always output something meaningful to users.

Comment (FGeri) [reply kornia/augmentation/random_generator/random_generator.py:358]: It's not really random stuff, since the function returns a tensor with shape [batch_size, 2], so returning one with shape of [0,2] for the `batch_size==0` case fits the pattern.

**Review by shijianjian — APPROVED** [review]
